package com.osttra;

import java.util.Scanner;

public class ExceptionHandling {

	public static void main(String[] args) {

		try {

			Scanner sc = new Scanner(System.in);

			System.out.println("Enter denominator!!");

			int denominator = sc.nextInt();

			int i = 4 / denominator;

			String name = "Rishika";

			System.out.println("Enter the index!!");
			int index = sc.nextInt();
			System.out.println(name.charAt(index));

		} catch (ArithmeticException e) {
			System.out.println("inside ArithmeticException catch...");
			e.printStackTrace();
			main(null);
		} catch (StringIndexOutOfBoundsException e) {
			System.out.println("inside StringIndexOutOfBoundsException catch...");
			e.printStackTrace();
			main(null);
		}
	}

	void askDenominator() {

		try {

			Scanner sc = new Scanner(System.in);

			System.out.println("Enter denominator!!");

			int denominator = sc.nextInt();

			int i = 4 / denominator;
		} catch (ArithmeticException e) {
			System.out.println("insdie catch of Arith");
		}
	}
	
	void askIndex() {
		
		try {
			
			Scanner sc = new Scanner(System.in);
			
			String name = "Rishika";

			System.out.println("Enter the index!!");
			int index = sc.nextInt();
			System.out.println(name.charAt(index));
		}
		catch (StringIndexOutOfBoundsException e) {
			System.out.println("inside catch of StringIndex...");
		}
	}
}

//public class ExceptionHandling {
//	
//	public static void main(String[] args) {
//		
//		try {
//			
//			Scanner sc = new Scanner(System.in);
//			
//			System.out.println("Enter denominator!!");
//			
//			int denominator = sc.nextInt();
//			
//			int i = 4/denominator;
//			
//			String name = "Rishika";
//			
//			System.out.println("Enter the index!!");
//			int index = sc.nextInt();
//			System.out.println(name.charAt(index));
//			
//		}catch (ArithmeticException e) {
//			System.out.println("inside ArithmeticException catch...");
//			e.printStackTrace();
//			main(null);
//		}
//		catch (StringIndexOutOfBoundsException e) {
//			System.out.println("inside StringIndexOutOfBoundsException catch...");
//			e.printStackTrace();
//			main(null);
//		}
//	}
//}

//public class ExceptionHandling {
//	
//	public static void main(String[] args) {
//		
//		try {
//			
//			Scanner sc = new Scanner(System.in);
//			
//			System.out.println("Enter denominator!!");
//			
//			int denominator = sc.nextInt();
//			
//			int i = 4/denominator;
//			
//			
//		}catch (Exception e) {
//			System.out.println("Please do not give 0 as denominator!!");
//			e.printStackTrace();
//			main(null);
//		}
//		System.out.println("second statement");	
//	}
//}
